trio/_core/_entry_queue.py

```diff
@@ -56,7 +56,13 @@ def run_cb(job):
                 async def kill_everything(exc):
                     raise exc
 
-                _core.spawn_system_task(kill_everything, exc)
+                try:
+                    _core.spawn_system_task(kill_everything, exc)
+                except RuntimeError:
+                    # We're quite late in the shutdown process and
+                    # the system nursery is already closed.
+                    _core.current_task().parent_nursery.start_soon(kill_everything, exc)
```

Comment:
This is a gross hack and I don't like it. (And I don't like the subtle coupling with the `finalize_remaining_asyncgens` code, where it has to use a shield because of this.)

But maybe there's no better solution for right now, and we should merge this as is, and then address #1607 in a separate PR.

Reply:
It is very much a gross hack, but I think it's the best solution for right now. I'll leave a comment.